tests/lolly/data/lolly_string_test.cpp

```diff
@@ -69,3 +69,14 @@ TEST_CASE ("test append") {
   str << string_u16 (u"yz");
   CHECK_EQ (str == string_u16 (u"xyz"), true);
 }
+
+TEST_CASE ("test reserve along with append") {
+  auto str= string_u16 ();
+  str->reserve (6);
```

Comment:
+ SUBCASE 1: reserve size > 0
+ SUBCASE 2: reserve size < 0
+ SUBCASE 3: reserve size = 0

we need better coverage of this interface

Reply:
n cannot be less than zero